nixos/modules/services/networking/oink.nix

```diff
@@ -20,14 +20,36 @@ in
   options.services.oink = {
     enable = lib.mkEnableOption "Oink, a dynamic DNS client for Porkbun";
     package = lib.mkPackageOption pkgs "oink" { };
+    apiKeyFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      example = "/run/keys/oink-api-key";
+      description = "Path to a file containing the API key to use when modifying DNS records.";
+    };
+    secretApiKeyFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      example = "/run/keys/oink-secret-api-key";
+      description = "Path to a file containing the secret API key to use when modifying DNS records.";
+    };
     settings = {
       apiKey = lib.mkOption {
         type = lib.types.str;
-        description = "API key to use when modifying DNS records.";
+        default = "";
+        description = ''
+          API key to use when modifying DNS records.
+
+          This option is discouraged as it makes the API key world-readable. Use {option}`apiKeyFile` instead.
+        '';
       };
       secretApiKey = lib.mkOption {
         type = lib.types.str;
-        description = "Secret API key to use when modifying DNS records.";
+        default = "";
+        description = ''
+          Secret API key to use when modifying DNS records.
+
+          This option is discouraged as it makes the API key world-readable. Use {option}`secretApiKeyFile` instead.
+        '';
```

Comment:
sorry, but I think these options should be definitely deprecated with `mkRemovedOptionModule`.
It is a trap people might fall into. I do not really want to encourage it.

By removing this the `assertions` are no longer necessary, the code gets simpler. The downside its a change you have to add to the release-notes.

if they want to use it this way the users could still use something like:

```nix
apiKeyFile = pkgs.writeText "api-key" "secret";
```